src/platform/systemtap.rs

```diff
@@ -181,7 +181,7 @@ macro_rules! sdt_asm(
 #[doc(hidden)]
 #[macro_export]
 macro_rules! _sdt_asm(
-    ($addr:tt, options $opt:tt, $provider:ident, $name:ident, $argstr:tt, $($arg:expr),*) => (
+    ($addr:tt, options ($opt:tt), $provider:ident, $name:ident, $argstr:tt, $($arg:expr),*) => (
```

Comment:
I think this change won't work for non-x86 targets that are passing just `options()` here. Maybe we should use a complete repetition pattern, `options ( $( $opt:ident ),* )`.

Reply:
You're right -- tested this on a non-x86 arch and it fails as you predict. Looking closer, though, do we actually *need* the `att_syntax`? If I remove it, it still builds for me on both 32- and 64-bit Linux x86, and as best I can tell everything we emit is either directives, which don't very with `gas` syntax, or a `nop`, which is perfectly valid Intel syntax. Removing that switch would certainly simplify things a bit.

Reply:
It matters for how the register is written in the probe note. When I first tried in the default (intel) syntax, GDB wasn't able to parse the probe argument correctly. Maybe GDB (and other tools!) could be taught otherwise, but then we'd only work with newer versions.